### DISCUSSION TITLE
http/dns

### DISCUSSION THREAD

**Discussion by giscus** [isno/theByteBook#181] [General]
# http/dns

构建大规模高可用的分布式系统

https://www.thebyte.com.cn/http/dns.html

**Comment by eastjoe** [comment]
"发出解析 thebyte.con.cn 域名" 、“向该地址请求 thebyte.con.cn. 的 Authoritative nameserver” 
应该是 thebyte.com.cn 域名吧

**Reply by isno** [reply]
域名打错了，谢谢。已修正。

**Comment by echo1937** [comment]
可以考虑把TLD nameserver的全称Top-Level Domains nameserver补充一下，看到TLD很懵

**Comment by hcjjj** [comment]
DNS 解析器” 判断是否存在解析缓存：
如存在，直接返回缓存的结果，也就是直接执行第 8 步；
如不存【在】，执行第 2 步，向就近的“根域名服务器”....

**Reply by hcjjj** [reply]
少了一个在字

**Reply by isno** [reply]
已修改，谢谢！

**Comment by dolphinlife** [comment]
"弄清楚解域名的结构"，多了一个解字

**Reply by isno** [reply]
已修改，谢谢！

**Comment by synthpop123** [comment]
技术服务商 Aakamai 的 Edge DNS 服务故障
应该是 Akamai，多了个 a

**Reply by isno** [reply]
谢谢，已修改！

**Comment by yuebaiqinghui** [comment]
您提供的https://root-servers.org/ 写的：The 13 root name servers are operated by 12 independent organisations.
As of 2025-10-21T01:43:04Z, the root server system consists of 1998 instances operated by the 12 independent root server operators.
这不就是说只有13个根域名服务器吗，你说的1845个是实例吧（现在是1998个）
“13台根服务器”是指13个逻辑上的根服务器标识（字母A-M）
1998个实例 = 这些13个根服务器在全球的物理/虚拟部署点（anycast节点），这些不是独立的根服务器，而是同一个根服务器的多个镜像

**Reply by lutasa** [reply]
域名根服务器一直都是13个，只是通过任播来做负载。文中的表述有误

**Comment by ic3Bing** [comment]
>“实际上，“DNS 解析器”也经常出现问题，这时候再使用 nslookup 命令就不行了。
当怀疑系统默认的“DNS 解析器”异常时，我们可以使用 dig 命令，通过切换不同的“DNS 解析器”，分析解析哪里出现异常。例如，使用 8.8.8.8 查询 thebyte.com.cn 的解析记录。”

这句话表达的意思看起来是 nslookup不支持切换除系统默认DNS解析器以外的localDNS地址，实际上与dig @x.x.x.x类似的是nslookup thebyte.com.cn x.x.x.x就能切换localDNS。

eg：

nslookup baidu.com 114.114.114.114
nslookup baidu.com 223.5.5.5

**Comment by ErickRen2023** [comment]
typo: 
in 2.3.1 域名解析的原理
"第 1 步，用户向“DNS 解析器”（Recursive resolver）发出解析 thebyte.con.cn 域名请求。“DNS 解析器”也称 LocalDNS，例如电信运营商的 114.114.114.114。"
thebyte.con.cn -> thebyte.com.cn
